shared/metrics/src/main/java/io/pravega/shared/metrics/MetricsConfig.java

```diff
@@ -209,4 +213,26 @@ public void setDynamicCacheEvictionDurationMs(long duration) {
     }
 
     //endregion
+
+    /**
+     * Tries to get the host name for this host. If the operation does not succeed, this method returns the legacy
+     * default value as a prefix for metrics (i.e., "pravega").
+     *
+     * @return Default hostname as a prefix for metrics reporters.
+     */
+    private static String getHostName() {
+        String hostname = "pravega";
+        try {
+            hostname = InetAddress.getLocalHost().getHostName();
+
+            // Depending on the environment, we may get the FQDN. In this case, we just get the leftmost part of it (hostname).
+            if (hostname.contains(".")) {
```

Comment:
I am always leery of hostname parsing code because I worry about edge cases like IPv6 addresses.   I think there's a non-zero chance of getting an IP address back from `getHostName()`.

Honestly this code seems unnecessary because the init script is going to set the value in most cases, based on the `HOSTNAME` environment variable which is preferable anyway.

Reply:
Thanks, I removed that. Let's rely on `HOSTNAME` envvar then.